### PR TITLE
Removed uppper restriction on sklearn version 

### PR DESCRIPTION
**Pull request by Piteryo**
New versions of sklearn have some great features like ColumnTransformer e.t.c. Tested your package on newer sklearn version - didn't find any issues.

### CODE REVIEW

**Comment by Piteryo** [comment]
Build fails due to assertion that fail because of new version

**Comment by mfeurer** [comment]
Thanks for the pull request. I'd be happy to migrate to the next version of scikit-learn, but unfortunately, it's not as easy as replacing the imports.

For an upgrade we would require:
* passing unit tests
* restrict the supported versions to 0.20.X
* go through the change log and implement novel methods and hyperparameters into Auto-sklearn
* Afterwards, we'd have to regenerate the meta data.

I'd be happy to help with the last bullet point and discuss/answer questions regarding the first three.

Also, could you please remove the c file from the pull request?

**Comment by amueller** [comment]
- Would you be happy to not support older versions?
- How do you determine which new methods to add, and why is that a requirement for making it compatible? ;)

having to fire up a separate venv for auto-sklearn that doesn't allow using current libraries is a bit of a bummer.

**Comment by amueller** [comment]
FYI 0.21 is on it's way ;)

**Comment by mfeurer** [comment]
> Would you be happy to not support older versions?

So far, auto-sklearn was only compatible with a single version of auto-sklearn at every point in time.

> How do you determine which new methods to add, and why is that a requirement for making it compatible? ;)

Hm, good question. There should be at least a brief discussion on what to update.

**Comment by amueller** [comment]
> So far, auto-sklearn was only compatible with a single version of auto-sklearn at every point in time.

I assume the second should be scikit-learn ;) And why is that? Because we keep renaming parameters? I feel like it shouldn't be too hard to make it work with multiple versions.
But mostly I feel making it work with the current version is nice. You can always tell people to upgrade their sklearn to install auto-sklearn but if you tell them to downgrade you're basically asking them to create a separate environment.

**Comment by mfeurer** [comment]
> And why is that? Because we keep renaming parameters? I feel like it shouldn't be too hard to make it work with multiple versions.

Easier maintenance.

> But mostly I feel making it work with the current version is nice. You can always tell people to upgrade their sklearn to install auto-sklearn but if you tell them to downgrade you're basically asking them to create a separate environment.

Agreed. That'll probably be the next big project after creating scripts for reproducing the results from the paper.

File: test/test_pipeline/implementations/test_imputation.py [hunk 1]
```diff
@@ -301,60 +301,60 @@ def test_imputation_copy(self):
...
-        imputer = Imputer(missing_values=0, strategy="mean", copy=False)
+        imputer = SimpleImputer(missing_values=0, strategy="mean", copy=False)
         Xt = imputer.fit(X).transform(X)
         Xt[0, 0] = -1
         assert_true(np.all(X == Xt))
 
         # copy=False, sparse csr, axis=1 => no copy
         X = X_orig.copy()
-        imputer = Imputer(missing_values=X.data[0], strategy="mean",
+        imputer = SimpleImputer(missing_values=X.data[0], strategy="mean",
```
Comment (Borda) [inline test/test_pipeline/implementations/test_imputation.py]: line alignment, PEP8

File: autosklearn/pipeline/components/data_preprocessing/imputation/imputation.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ def fit(self, X, y=None):
         # Imputation does not support fit_transform (as of 0.19.1)!
         import sklearn.preprocessing
 
-        self.preprocessor = sklearn.preprocessing.Imputer(
+        self.preprocessor = sklearn.impute.SimpleImputer(
```
Comment (Borda) [inline autosklearn/pipeline/components/data_preprocessing/imputation/imputation.py]: probably single line

**Comment by mfeurer** [comment]
Thanks a lot for this PR, but this is now superseded by #703 (which actually extends this PR).

**Pull request closed without merging**